src/settings/settings.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/sequelize';
+import { Settings } from './settings.model';
+
+@Injectable()
+export class SettingsService {
+  constructor(
+    @InjectModel(Settings)
+    private settingsRepository: typeof Settings,
+  ) {}
+
+  async createSetting(settingData: Settings): Promise<Settings> {
+    await this.validateDataType(settingData);
+    return await this.settingsRepository.create(settingData);
+  }
+
+  async updateSetting(id: number, settingData: Settings): Promise<Settings> {
+    const existingSetting = await this.getExistingSetting(id);
+
+    if (settingData.data_type !== null) {
+      existingSetting.data_type = settingData.data_type;
+    }
+    if (settingData.value !== null) {
+      existingSetting.value = settingData.value;
+    }
+    if (settingData.name !== null) {
+      existingSetting.name = settingData.name;
+    }
+    existingSetting.updatedAt = new Date();
```

Comment:
We won't have to do this manually if we add proper decorators for `@UpdatedAt` in the sequelize model definition.